woocommerce/woocommerce
# Migrate Daily smoke tests to Playwright

### All Submissions:


### Changes proposed in this Pull Request:

This PR migrates our "Smoke test daily" workflow to using Playwright tests. The specific changes are:

- Adding of the `update-woocommerce.spec.js` test, which is the Playwright version of `update-woocommerce.js`.
- Adding of the `upload-plugin.spec.js` test, which is the Playwright version of `upload-plugin.js`.
- Adding of the new `plugins/woocommerce/tests/e2e-pw/utils` folder where we will migrate the utility functions from `e2e-environment`.
- Uploading of Allure reports as artifacts in the `smoke-test-daily.yml` workflow.
- Uploading of these Allure reports to the S3 bucket.
- Reporting of E2E stats in Slack.
- Posting of test summary of API and E2E tests on trunk at the end of the `smoke-test-daily.yml` workflow.

I have added PR review comments in the "Files changed" section to help in clarifying specific changes done in this PR.

Closes the following tasks:

- https://github.com/woocommerce/woocommerce-quality/issues/331
- https://github.com/woocommerce/woocommerce-quality/issues/332
- https://github.com/woocommerce/woocommerce-quality/issues/333
- https://github.com/woocommerce/woocommerce-quality/issues/334

### How to test the changes in this Pull Request

1. Go to the [Smoke test daily workflow runs page](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml).
2. In the "Run workflow" dropdown, select this branch (`e2e/daily-tests-playwright`) and click "Run workflow".
   <img width="264" alt="juQnNwYGSh" src="https://user-images.githubusercontent.com/4509348/196176252-8bfedb30-56c1-43ea-b14a-b7dbb213aa87.png">
3. A new "Smoke test daily" workflow run will be initiated. Open it, and wait for the workflow to finish.
4. Verify that the workflow run looks like this when it's finished.
   <img width="1264" alt="yZ4SiwlDaX" src="https://user-images.githubusercontent.com/4509348/196176450-05aace00-c359-4882-877a-b2a97e496ebf.png">
5. Scroll down the page a bit. You should see a list of artifacts for each of the E2E & API tests that ran. These are the Allure reports generated by each test.
   <img width="676" alt="a5k0uPvahY" src="https://user-images.githubusercontent.com/4509348/196176566-d3c06b33-10e9-4965-b23a-9ac34af22d24.png">
6. Scroll down further. You should see the test summary for "Smoke tests on trunk" like so:
   <img width="746" alt="XkJ8QBAtsE" src="https://user-images.githubusercontent.com/4509348/196176705-51e9bfbd-3e06-4792-ae19-3a50996f340d.png">
7. Make sure that the links in the "Smoke tests on trunk" summary are working:
   - _To view the full API test report, click here_ should take you to https://woocommerce.github.io/woocommerce-test-reports/daily/api.
   - _To view the full E2E test report, click here_ should take you to https://woocommerce.github.io/woocommerce-test-reports/daily/e2e.
   - _To view all test reports, visit the WooCommerce Test Reports Dashboard._ should take you to https://woocommerce.github.io/woocommerce-test-reports
8. Make sure that when you go to the following URL's, you're taken to the Allure report of the corresponding plugin (you should see the plugin name being mentioned on the title of the Allure report):
    <img width="531" alt="UoLTFcLo6I" src="https://user-images.githubusercontent.com/4509348/196178575-eec4dd54-944e-4cbe-8594-dd2d0627134b.png">
   - WooCommerce Payments: https://a8c-woo-test-reports.s3.amazonaws.com/public/daily/woocommerce-payments/e2e/index.html
   - WooCommerce PayPal Payments: https://a8c-woo-test-reports.s3.amazonaws.com/public/daily/woocommerce-paypal-payments/e2e/index.html
   - WooCommerce Shipping & Tax: https://a8c-woo-test-reports.s3.amazonaws.com/public/daily/woocommerce-services/e2e/index.html
   - WordPress SEO: https://a8c-woo-test-reports.s3.amazonaws.com/public/daily/woocommerce-payments/e2e/index.html
   - Contact Form 7: https://a8c-woo-test-reports.s3.amazonaws.com/public/daily/contact-form-7/e2e/index.html

The test summary for the "Smoke tests with ${PLUGIN} plugin installed" will be added in a later PR.

### Other information:




### FOR PR REVIEWER ONLY:
